dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-details-by-id.controller/helpers/parse-filters.ts

```diff
@@ -0,0 +1,22 @@
+import { UtilisationReportPremiumPaymentsFilters } from '@ukef/dtfs2-common';
+import { REGEX } from '../../../../../constants';
+
+/**
+ * Parses the provided premium payments tab filters to test validity.
+ * @param premiumPaymentsFilters - The unparsed filters to be applied for the premium payments table
+ * @param premiumPaymentsFilters.facilityId - The facility ID filter
+ * @returns The parsed filters for the premium payments tab
+ */
+export const parsePremiumPaymentsFilters = (premiumPaymentsFilters?: UtilisationReportPremiumPaymentsFilters): UtilisationReportPremiumPaymentsFilters => {
+  if (!premiumPaymentsFilters) {
+    return {};
+  }
+
+  const { facilityId } = premiumPaymentsFilters;
+
+  const facilityIdFilter = facilityId && REGEX.UKEF_PARTIAL_FACILITY_ID_REGEX.test(facilityId) ? facilityId : undefined;
```

Comment:
```suggestion
  const facilityIdFilter = (facilityId && REGEX.UKEF_PARTIAL_FACILITY_ID_REGEX.test(facilityId)) && facilityId;
```

Reply:
This would return `false` when `facilityId` is defined but doesn't match the regex:

Sample standalone code to return `false`:
```
const PARTIAL_FACILITY_ID_REGEX = /^\d{4,10}$/;

const facilityId = 'abc1';
const facilityIdFilter = (facilityId && PARTIAL_FACILITY_ID_REGEX.test(facilityId)) && facilityId;

console.log(facilityIdFilter);
```

Reply:
@sw-mattw How about this instead?

```ts
if (!premiumPaymentsFilters || !premiumPaymentsFilters.facilityId) {
  return {};
}

const { facilityId } = premiumPaymentsFilters;

let facilityIdFilter;

if (REGEX.UKEF_PARTIAL_FACILITY_ID_REGEX.test(facilityId)) {
  facilityIdFilter = facilityId;  
}

return {
  facilityId: facilityIdFilter,
};
```

I know there's more lines, but feels clearer than having a ternary explicitly returning `undefined`?

Reply:
Apologies, just pushed that